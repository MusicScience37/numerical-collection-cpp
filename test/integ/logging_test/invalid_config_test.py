"""Test of invalid logging configuration files.
"""

import pathlib
import subprocess

import approvaltests

from .create_scrubber import create_scrubber


def _verify_command_result(
    writer_path: pathlib.Path,
    config_file_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
) -> None:
    result = subprocess.run(
        args=[str(writer_path), str(config_file_path)],
        cwd=str(test_temp_dir_path),
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        check=False,
        encoding="utf-8",
    )

    approvaltests.verify(
        f"""exit code: {result.returncode}
stdout:
{result.stdout}
stderr:
{result.stderr}
""",
        options=approvaltests.Options().with_scrubber(
            create_scrubber(str(config_file_path))
        ),
    )


def test_directory(
    writer_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
    config_dir_path: pathlib.Path,
):
    """Test to pass directory paths."""

    _verify_command_result(
        writer_path=writer_path,
        config_file_path=config_dir_path,
        test_temp_dir_path=test_temp_dir_path,
    )


def test_nonexisting_file(
    writer_path: pathlib.Path,
    test_temp_dir_path: pathlib.Path,
    config_dir_path: pathlib.Path,
):
    """Test to pass non-existing paths."""

    _verify_command_result(
        writer_path=writer_path,
        config_file_path=config_dir_path / "non_existing_file.toml",
        test_temp_dir_path=test_temp_dir_path,
    )
