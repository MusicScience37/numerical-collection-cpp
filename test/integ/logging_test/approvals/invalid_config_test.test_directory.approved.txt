exit code: 1
stdout:

stderr:
Exception thrown: Invalid filepath to load configurations <config-filepath> (load_logging_config.h:42:<col>)

